src/refract/primitives.es6

```diff
@@ -0,0 +1,322 @@
+/*
+ * A refract element implementation with an extensible type registry.
```

Comment:
@smizell I've vendored this for the time being because it feels wrong to open a pull request without updating the tests :tired_face:. I want this to be temporary until minim.js is updated, which will still require a good bit of work. My changes:

* [x] Convert to ES6
* [x] Remove `PropertyType`
* [x] Add `meta` to elements
* [x] Add a `fromCompactRefract` to support both input formats
* [x] Renamed a few variables to help non-native English speakers / readability (e.g. `val` -> `value`)
* [x] Added a global type registry (see `src/refract/api.es6` in this PR for how it is used)

Remaining work:
* [ ] Support refracted elements within `attributes`, e.g. `transaction.request.headers`
* [ ] Better error handling

Open questions:
* [ ] Should `ArrayType().push` take in refracted type instances or plain js values?
* [ ] Should `ObjectType().set` take in refracted type instances or plain js values?